### PR TITLE
import numpy as np

### PR DESCRIPTION
**Pull request by kvmanohar22**
Changed all the numpy imports to a single form of `import numpy as np`
fix gh-4218

### CODE REVIEW

**Comment by nouiz** [comment]
Travis have some failure. But they don't seem hard to fix. Can you fix them quickly?

I forgot to tell that for that type of change, normally we should do I commit per file and about one PR per directory. As I wrote that too late, the second best option is to try to finish this the fastest, as otherwise there will be very rapidly conflict with other PR. This is why I asked if you can finish them quickly.

To help you fix some of travis failure, run locally that command line to help make sure one test is well fixed. Don't rely on travis for this. It will slow down the fix of it:

    nosetests theano/tests/test_flake8.py

thanks

**Comment by kvmanohar22** [comment]
Yes sure, I'll fix them soon

File: theano/tensor/tests/test_opt.py [hunk 1]
```diff
@@ -6649,6 +6649,81 @@ def test_local_useless_alloc():
     assert isinstance(topo[-1].op, T.Alloc)
 
 
+def compile_graph_log_sum_exp(x, axis, dimshuffle_op=None):
```
Comment (nouiz) [inline theano/tensor/tests/test_opt.py]: Why this is added? I don't think it should be added by this PR.

File: theano/tensor/tests/test_opt.py [hunk 2]
```diff
@@ -377,19 +377,19 @@ def test_elemwise_multiple_inputs_optimisation(self):
 #            (dz*fy*(fx+fy),(fx,fy,dz),(dxv,dyv,dzv),2,'float64'),#check mixed type mul
             # check with dimshuffle of constant
             (fx + fy + fz + 2, (fx, fy, fz), (fxv, fyv, fzv), 1, {'custom':
-                 'float32', 'numpy+floatX': config.floatX, 'numpy': 'float64'}),
+                 'float32', 'np+floatX': config.floatX, 'np': 'float64'}),
```
Comment (nouiz) [inline theano/tensor/tests/test_opt.py]: Why do you revert all the fix you did in the previous commit? This is string, we don't want to change them to np. The issue was just about changing the code.

**Comment by nouiz** [comment]
travis have failure. I think it is best that you fix travis first, then rebase as this PR now have conflict.

File: benchmark/autoencoder/aa.py [hunk 3]
```diff
@@ -13,7 +13,7 @@
 
 from theano.tensor import blas, opt
 
-# numpy: aa_numpy.py
```
Comment (lamblin) [inline benchmark/autoencoder/aa.py]: Please do not blindly auto-replace "numpy" by "np". Here, we are talking about a numpy implementation and a filename, not about the module name.
You could replace "numpy:" in that comment by "NumPy" if you want, but "np" is not clear enough.
Moreover, the name of the file is still "aa_numpy.py", not "aa_np.py".

File: benchmark/numexpr/gen_graph.py [hunk 4]
```diff
@@ -31,7 +31,7 @@ def timeit_2vector_theano(init, nb_element=1e6, nb_repeat=3, nb_call=int(1e2), e
 
 def timeit_2vector(nb_element=1e6, nb_repeat=3, nb_call=int(1e2), expr="a**2 + b**2 + 2*a*b", do_unalign=False, do_amd=True):
     """Returns a dictionary whose keys are implementations ('numpy', 'numexpr', 'theano', etc.)
-    and whose values are numpy arrays of times taken to evalute the given problem.
```
Comment (lamblin) [inline benchmark/numexpr/gen_graph.py]: Keep "numpy" or NumPy" in text.

File: doc/acknowledgement.txt [hunk 5]
```diff
@@ -14,7 +14,7 @@ Acknowledgements
 * All `Theano contributors <https://github.com/Theano/Theano/graphs/contributors>`_.
 * All Theano users that have given us feedback.
 * The GPU implementation of tensordot is based on code from Tijmen
-  Tieleman's `gnumpy <http://www.cs.toronto.edu/~tijmen/gnumpy.html>`_
```
Comment (lamblin) [inline doc/acknowledgement.txt]: No, the package name is "gnumpy", not "gnp".

File: doc/cifarSC2011/gpundarray.txt [hunk 6]
```diff
@@ -12,7 +12,7 @@ Why a common GPU ndarray?
   - CudaNdarray (Theano), GPUArray (PyCUDA), CUDAMatrix (cudamat), GPUArray (PyOpenCL), ...
   - There are even more if we include other languages
 
-- All of them are a subset of the functionality of ``numpy.ndarray`` on the GPU
```
Comment (lamblin) [inline doc/cifarSC2011/gpundarray.txt]: This is text, not code, so I would keep ``numpy.ndarray``

File: doc/cifarSC2011/gpundarray.txt [hunk 7]
```diff
@@ -24,7 +24,7 @@ Why a common GPU ndarray?
 
 Design Goals
 
-- Make it VERY similar to ``numpy.ndarray``
```
Comment (lamblin) [inline doc/cifarSC2011/gpundarray.txt]: same here.
Actually, these are old tutorials that got archived here, it is not really necessary to update them.

File: doc/conf.py [hunk 8]
```diff
@@ -195,8 +195,8 @@ def setup(app):
 # This function is used to populate the (source) links in the API
 def linkcode_resolve(domain, info):
     def find_source():
-        # try to find the file and line number, based on code from numpy:
-        # https://github.com/numpy/numpy/blob/master/doc/source/conf.py#L286
```
Comment (lamblin) [inline doc/conf.py]: Keep "numpy" in the url and the text

File: doc/crei2013/gpundarray.txt [hunk 9]
```diff
@@ -13,7 +13,7 @@ Why a common GPU ndarray?
   - CudaNdarray (Theano), GPUArray (PyCUDA), CUDAMatrix (cudamat), GPUArray (PyOpenCL), ...
   - There are even more if we include other languages
 
-- All of them are a subset of the functionality of ``numpy.ndarray`` on the GPU
```
Comment (lamblin) [inline doc/crei2013/gpundarray.txt]: Same here (it is the same text as before).

File: doc/extending/extending_theano.txt [hunk 10]
```diff
@@ -847,26 +847,26 @@ as_op Example
...
+    f = function([x, y], np_dot(x, y))
+    inp1 = np.random.rand(5, 4).astype('float32')
+    inp2 = np.random.rand(4, 7).astype('float32')
     out = f(inp1, inp2)
 
 
 Exercise
 ^^^^^^^^
 
-Run the code of the *numpy_dot* example above.
```
Comment (lamblin) [inline doc/extending/extending_theano.txt]: Still "numpy_dot", which is the name of the function defined above.

File: doc/extending/extending_theano_gpu.txt [hunk 11]
```diff
@@ -124,7 +125,7 @@ instead.  It adds a new section to the parsed files whose tag is
 ``#kernel name:params:flags``.
 
 Here ``name`` is the name of the kernel function in the following
-code, ``params`` is a comma-separeted list of numpy typecode names.
+code, ``params`` is a comma-separeted list of np typecode names.
```
Comment (lamblin) [inline doc/extending/extending_theano_gpu.txt]: "numpy" or "NumPy"

File: doc/extending/extending_theano_solution_1.py [hunk 12]
```diff
@@ -179,8 +179,8 @@ def infer_shape_numpy_dot(node, input_shapes):
 
 @as_op(itypes=[theano.tensor.fmatrix, theano.tensor.fmatrix],
        otypes=[theano.tensor.fmatrix], infer_shape=infer_shape_numpy_dot)
-def numpy_add(a, b):
-    return numpy.add(a, b)
+def np_add(a, b):
```
Comment (lamblin) [inline doc/extending/extending_theano_solution_1.py]: I would keep the old name "numpy_add", it is more clear, and the infer shape function is still named "infer_shape_numpy_add_sub".

File: doc/extending/fibby.txt [hunk 13]
```diff
@@ -77,8 +77,8 @@ TensorType also set up ``dtype_%(x)s`` to be a typdef to the C type for ``x``.
 The first line reduces the reference count of the data that y originally
 pointed to. The second line allocates the new data and makes y point to it.
 
-In C code for a theano op, numpy arrays are represented as ``PyArrayObject`` C
-structs. This is part of the numpy/scipy C API documented at
+In C code for a theano op, np arrays are represented as ``PyArrayObject`` C
+structs. This is part of the np/scipy C API documented at
```
Comment (lamblin) [inline doc/extending/fibby.txt]: Keep "numpy" or "NumPy" in the text

File: doc/extending/inplace.txt [hunk 14]
```diff
@@ -7,7 +7,7 @@ Views and inplace operations
 
 Theano allows the definition of Ops which return a :term:`view` on one
 of their inputs or operate :term:`inplace` on one or several
-inputs. This allows more efficient operations on numpy's ``ndarray``
+inputs. This allows more efficient operations on np's ``ndarray``
```
Comment (lamblin) [inline doc/extending/inplace.txt]: "numpy" or "NumPy"

File: doc/extending/type.txt [hunk 15]
```diff
@@ -50,7 +50,7 @@ default values.
       of this writing this is used only when we transfer new data to a
       shared variable on the gpu.
 
-      ``storage`` will be the old value. i.e. The old numpy array,
+      ``storage`` will be the old value. i.e. The old np array,
```
Comment (lamblin) [inline doc/extending/type.txt]: "numpy" or "NumPy"

File: doc/extending/unittest.txt [hunk 16]
```diff
@@ -222,7 +222,7 @@ Checking for correctness
 ------------------------
 
 When checking for correctness of mathematical expressions, the user
-should preferably compare theano's output to the equivalent numpy
+should preferably compare theano's output to the equivalent np
```
Comment (lamblin) [inline doc/extending/unittest.txt]: Here as well

File: doc/extending/unittest.txt [hunk 17]
```diff
@@ -355,7 +355,7 @@ do the following:
 
 The behaviour of seed_rng is as follows:
 
-* If an explicit seed is given, it will be used for seeding numpy's rng.
+* If an explicit seed is given, it will be used for seeding np's rng.
```
Comment (lamblin) [inline doc/extending/unittest.txt]: numpy or NumPy

File: doc/glossary.txt [hunk 18]
```diff
@@ -24,8 +24,8 @@ Glossary
         the dimensions that it is lacking.  
         
         For more detail, see :ref:`libdoc_tensor_broadcastable`, and also
-        * `SciPy documentation about numpy's broadcasting <http://www.scipy.org/EricsBroadcastingDoc>`_
-        * `OnLamp article about numpy's broadcasting <http://www.onlamp.com/pub/a/python/2000/09/27/numerically.html>`_
+        * `SciPy documentation about np's broadcasting <http://www.scipy.org/EricsBroadcastingDoc>`_
+        * `OnLamp article about np's broadcasting <http://www.onlamp.com/pub/a/python/2000/09/27/numerically.html>`_
```
Comment (lamblin) [inline doc/glossary.txt]: numpy or NumPy

File: doc/index.txt [hunk 19]
```diff
@@ -198,7 +198,7 @@ address facet Y" and link to the previous discussion.
 
 When asking questions on StackOverflow, please use the *theano* tag, so your question can be found, and follow
 StackOverflow's guidance on `asking questions <http://stackoverflow.com/help/asking>`__. Consider also using the
-*python* and *numpy* tags, especially if you are unsure which library your problem relates to.
+*python* and *np* tags, especially if you are unsure which library your problem relates to.
```
Comment (lamblin) [inline doc/index.txt]: The tag on StackOverflow is named "numpy"

**Comment by lamblin** [comment]
Also, having everything in one giant commit goes beyond the capacity of GitHub's interface.
When you rebase, could you also split your changes into different commits, maybe one for each major directory in `theano/` (`tensor/`, `sparse/`) and one for `doc/`?

**Comment by kvmanohar22** [comment]
I'm doing this over again, I made a new branch and I'm doing one commit per file and one PR per directory as in `theano/ (tensor/, sparse/)` and one for `doc/`

**Comment by lamblin** [comment]
Thanks.

**Comment by nouiz** [comment]
replaced by gh-5179, so closing.

**Pull request closed without merging**